pkg/agent/agent.go

```diff
@@ -206,7 +165,54 @@ func (k *Klocksmith) Run() error {
 	k.lc.Reboot(false)
 
 	// cross fingers
-	time.Sleep(24 * 7 * time.Hour)
+	sleepOrDone(24*7*time.Hour, stop)
+	return nil
```

Comment:
Solving #93 depends on handling the termination signal by closing stop, and thus goroutines and sleeps, like in the above example. It can't depend on Run returning.  From the cleanup angle, neither sleep and sleepOrDone (where currently stop won't be done) are sufficient.